### PR TITLE
add ffmpeg builders

### PR DESCRIPTION
**Pull request by SimonDanisch**
I added an `add_builder` script for this purpose... I can remove it, but maybe it's helpful to also upgrade other builders?
This is mostly auto generated, so will need to review this PR a bit!

### CODE REVIEW

File: L/Libass/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    "https://github.com/libass/libass/releases/download/0.14.0/libass-0.14.0.tar.xz" =>
+    "881f2382af48aead75b7a0e02e65d88c5ebd369fe46bc77d9270a94aa8fd38a2",
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+apk add nasm
+cd libass-0.14.0/
+sed -i 's/9.10.3/2.9.1/' configure.ac
```
Comment (giordano) [inline L/Libass/build_tarballs.jl]: This hack and following autoreconf should not be necessary anymore. Probably also setting AR will be unnecessary

File: F/FFMPEG/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  --enable-cross-compile \
+  --cross-prefix=/opt/${target}/bin/${target}- \
+  --arch=${ccARCH}     \
+  --target-os=${ccOS}  \
+  --cc="${CC}"         \
+  --cxx="${CXX}"       \
+  --dep-cc="${CC}"     \
+  --ar="${AR}"         \
+  --nm="${NM}"         \
+  --objcc="${CC} -ObjC" \
```
Comment (giordano) [inline F/FFMPEG/build_tarballs.jl]: There is the OBJC environment variable now

File: L/Libvpx/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+name = "LibVPX"
+version = v"1.8.0"
+
+# Collection of sources required to build LibVPX
+sources = [
+    "https://github.com/webmproject/libvpx/archive/v1.8.0.tar.gz" =>
+    "86df18c694e1c06cc8f83d2d816e9270747a0ce6abe316e93a4f4095689373f6",
+
+    "./patches"
```
Comment (giordano) [inline L/Libvpx/build_tarballs.jl]: The directory is not here 🙂

File: X/X265Builder/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd x265_3.0/
+apk add nasm
+export PKG_CONFIG_PATH="${prefix}/lib/pkgconfig"
+mkdir bld && cd bld
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=/opt/$target/$target.toolchain -DENABLE_PIC=ON -DENABLE_SHARED=off ../source
```
Comment (giordano) [inline X/X265Builder/build_tarballs.jl]: Now there is an environment variable that points to the location of the cmake toolchain file (cannot easily check its name on my phone)

File: L/Libass/build_tarballs.jl (hunk 1 above)
Comment (SimonDanisch) [reply L/Libass/build_tarballs.jl]: Can you directly remove those (github allows to edit files)? I don't really understand these hacks, so the probability that I mess something up while applying your comment is quite high ;)

File: X/X265Builder/build_tarballs.jl (hunk 4 above)
Comment (giordano) [reply X/X265Builder/build_tarballs.jl]:
```suggestion
cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE="${CMAKE_TARGET_TOOLCHAIN}" -DENABLE_PIC=ON -DENABLE_SHARED=off ../source
```

File: L/Libass/build_tarballs.jl (hunk 1 above)
Comment (giordano) [reply L/Libass/build_tarballs.jl]: I don't think I can edit the pull request.  My suggestion is to simply delete lines 17 and 18 (these lines were working around an issue with the FreeType pkgconfig file, now fixed with https://github.com/JuliaPackaging/Yggdrasil/commit/d2297a46ccf9e9983028bdee8780e658d48e9f75)

Comment (SimonDanisch) [reply L/Libass/build_tarballs.jl]: Didn't you just suggest an edit? :D

Comment (giordano) [reply L/Libass/build_tarballs.jl]: Ok, I thought you were suggesting me to push commits to the pull request :sweat_smile: Here you literally only need to delete two lines, it's a bit annoying that you can't suggest multi-line changes.

**Comment by staticfloat** [comment]
Hey Simon, could you split these up into slightly smaller PRs?  I will be happy to start with the simpler ones, like `X265` and `libVPX`, then work my way up to FFMPEG.  :)

**Comment by haberdashPI** [comment]
How can I help to move this along? I'm happy to start breaking out some of the individual dependencies into individual PR's.

**Comment by giordano** [comment]
I've removed from this PR `x265` and `libVPX`, then I cleaned up the other builders, and tested locally those for which all dependencies are already available

**Comment by haberdashPI** [comment]
Great! Do those remaining dependencies need to be split up into separate PR's? Can I help?

**Comment by staticfloat** [comment]
@haberdashPI Yes, if you can take individual builders one by one and open PRs with each of them, that would help me.  :)

**Pull request merged**